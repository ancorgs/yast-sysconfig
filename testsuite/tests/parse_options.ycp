// Sysconfig module - testsuite
//
// ParseString tests
//
// $Id$
//

{

include "testsuite.ycp";

import "Pkg";   // override packamanager
import "Sysconfig";

map param = $[ "separator" : ",",
    "unique" : true,
    "remove_whitespace" : true
];

// Basic functionality
TEST( ``(Sysconfig::ParseString(nil, param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString("", param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString("    ", param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString(" abc , 123, var", param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString(" abc,123  ", param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString("x,  y  ,  z", param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString("0,\"1, 2, 3\", 4, 5 ", param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString(" ,,  ,", param) ), [ $[], $[], $[] ], nil );

// multi separator
param["separator"] = ",;.";
TEST( ``(Sysconfig::ParseString("ab.sdf;  fwre  sdf", param) ), [ $[], $[], $[] ], nil );

// uniqe / multile test
TEST( ``(Sysconfig::ParseString(" 1,2,1,1,5 ,", param) ), [ $[], $[], $[] ], nil );
param["unique"] = false;
TEST( ``(Sysconfig::ParseString(" 1,2,1,1,5 ,", param) ), [ $[], $[], $[] ], nil );

// use space as separator
param["separator"] = " ";
TEST( ``(Sysconfig::ParseString(" \"\\\"\", 2", param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString("   a   b zx ,, ,", param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString("   123\"56\"78  bzx ", param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString("   a \\\\ b ", param) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::ParseString(" \n \\\"  ", param) ), [ $[], $[], $[] ], nil );

// don't remove white space
param["remove_whitespace"] = false;
param["separator"] = ",";
TEST( ``(Sysconfig::ParseString("   a   b zx ,, ,", param) ), [ $[], $[], $[] ], nil );

// comma as separator
param["separator"] = ",";
TEST( ``(Sysconfig::ParseString(" ,,  ,", param) ), [ $[], $[], $[] ], nil );
param["remove_whitespace"] = true;
TEST( ``(Sysconfig::ParseString(" ,,  ,", param) ), [ $[], $[], $[] ], nil );
// don't remove empty strings
param["remove_empty_strings"] = false;
TEST( ``(Sysconfig::ParseString(" ,,  ,", param) ), [ $[], $[], $[] ], nil );
}

