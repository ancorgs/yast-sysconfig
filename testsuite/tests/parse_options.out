Return	[]
Return	[]
Return	[]
Return	["abc", "123", "var"]
Return	["abc", "123"]
Return	["x", "y", "z"]
Return	["0", "1, 2, 3", "4", "5"]
Return	[]
Return	["ab", "sdf", "fwre  sdf"]
Return	["1", "2", "5"]
Return	["1", "2", "1", "1", "5"]
Return	["\"", ",", "2"]
Return	["a", "b", "zx", ",,", ","]
Return	["123\"56\"78", "bzx"]
Return	["a", "\\", "b"]
Return	["\n", "\""]
Return	["   a   b zx ", " "]
Return	[" ", "  "]
Return	[]
Return	["", ""]
